src/shared-endpoints.ts

```diff
@@ -0,0 +1,95 @@
+type Constructor = new (...args: any[]) => {};
+import assert from 'assert';
+import { GenericAPIResponse } from './util/requestUtils';
+import RequestWrapper from './util/requestWrapper';
+
+export function SharedEndpoints<TBase extends Constructor>(Base: TBase) {
+  return class Shared extends Base {
```

Comment:
See comments at end of linear-client, this could just be `export class SharedEndpoints`, so linear and rest could just extend this